ui/page/main_page.go

```diff
@@ -218,20 +226,51 @@ func (mp *MainPage) OnNavigatedTo() {
 			go mp.WL.MultiWallet.Politeia.Sync()
 		}
 	}
+
+	mp.UpdateBalance()
 }
 
 func (mp *MainPage) setLanguageSetting() {
 	langPre := mp.WL.Wallet.ReadStringConfigValueForKey(load.LanguagePreferenceKey)
 	values.SetUserLanguage(langPre)
 }
 
-func (mp *MainPage) UpdateBalance(fetchExch bool) {
-	if len(mp.totalBalanceUSD) == 0 || fetchExch {
-		load.GetUSDExchangeValue(&mp.dcrUsdtBittrex)
-	}
+func (mp *MainPage) updateExchangeSetting() {
 	currencyExchangeValue := mp.WL.Wallet.ReadStringConfigValueForKey(dcrlibwallet.CurrencyConversionConfigKey)
-	mp.usdExchangeSet = currencyExchangeValue == values.USDExchangeValue
+	usdExchangeSet := currencyExchangeValue == values.USDExchangeValue
+	if mp.usdExchangeSet == usdExchangeSet {
+		return // nothing has changed
+	}
+	mp.usdExchangeSet = usdExchangeSet
+	if mp.usdExchangeSet {
+		go mp.fetchExchangeRate()
+	}
+}
 
+func (mp *MainPage) fetchExchangeRate() {
+	if mp.isFetchingExchangeRate {
+		return
+	}
+	maxAttempts := 7
```

Comment:
In testing, this is quite a long time. I've experimented with 3 tries and that seems okay. 3 tries will wait for a total of 6 seconds, aside the time each network call attempt takes. Can experiment with a different figure and see what is better for UX. While we want to try repeatedly, user should know soon enough when it isn't working so they can check their connection. Applies to send page as well.